src/librustc_typeck/astconv.rs

```diff
@@ -595,7 +590,9 @@ impl<'o, 'gcx: 'tcx, 'tcx> dyn AstConv<'gcx, 'tcx> + 'o {
             infer_types,
         );
 
-        let is_object = self_ty.map_or(false, |ty| ty.sty == TRAIT_OBJECT_DUMMY_SELF);
+        let is_object = self_ty.map_or(false, |ty| {
+            ty.sty == self.tcx().gcx.types.trait_object_dummy_self.sty
```

Comment:
```suggestion
            ty == self.tcx().gcx.types.trait_object_dummy_self
```

Reply:
Is comparing `Ty == Ty` the same as comparing `TyKind == TyKind`? I intentionally kept the `.sty`, since the `PartialEq` impl of `Ty` compares the pointers, while the `PartialEq` impl for `TyKind` is derived.

Reply:
It definitely makes no difference here (and should not make a difference anywhere else, that would be a bug).